Installer.cpp

```diff
@@ -129,6 +129,17 @@ void inline CleanupHack()
 
 HRESULT MoveFileToTempAndScheduleDeletion(const wstring& filePath, bool moveToTempDirectory)
 {
+    // Recommended way to check if a file exist:
+    // https://devblogs.microsoft.com/oldnewthing/20071023-00/?p=24713
+    DWORD fileAttributes = GetFileAttributesW(filePath.data());
```

Comment:
Why don't we use [PathFileExistsW](https://learn.microsoft.com/fr-fr/windows/win32/api/shlwapi/nf-shlwapi-pathfileexistsw) API, so it's even more certain and explicite?

Reply:
And use rather `filePath.c_str()` instead of `filePath.data()` - the reason is that c_str() ensures to return a c string with NULL pointer at the end. Whereas data() depends on the implementation.
Ref:
https://stackoverflow.com/questions/194634/string-c-str-vs-data

Reply:
The reason for using the function is that is what Microsoft recommend (Raymond Chen has been part of the Windows team at Microsoft for 30+ years).

So my thinking was, that if it is good enough for the Windows team itself, it should be fine to use 😃

Reply:
I will update the string thing tomorrow

Reply:
@GurliGebis 
I guess you're talking about this article:
https://devblogs.microsoft.com/oldnewthing/20071023-00/?p=24713

@xomx What do you think about PathFileExistsW vs. GetFileAttributesW ?

Reply:
@donho 
Definitely use the GetFileAttributes way here. 

Not only because of what this article says about possible network "deceptions" regarding the existence of a file. I have already explained (PathFileExists does not only check for a file existence but for a file system object (== file or folder) existence...) why the generally used PathFileExists in the N++ source code should be replaced by two different funcs but it was rejected:
https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14839#issuecomment-1988154458